misc/cf-support

```diff
@@ -0,0 +1,168 @@
+.SH SYNOPSIS
+.B cf-support
+.RI [ OPTION ]
+.SH DESCRIPTION
+cf-support gathers various details about the system and creates a tarball in the current directory to submit to support. If the system is an enterprise hub then additional details will be gathered and included. The utility will prompt for an optional support ticket number as well as prompt whether to include masterfiles in the tarball.
+
+.SH OPTIONS
+.IP "--yes"
+Non-interactive use. Assume no ticket number and assume include masterfiles.
+.SH CFENGINE
+CFEngine provides automated configuration management of large-scale computer systems. A system administrator describes the desired state of a system using CFEngine policy code. The program \fBcf-agent\fR reads policy code and attempts to bring the current system state to the desired state described. Policy code is downloaded by \fBcf-agent\fR from a \fBcf-serverd\fR daemon. The daemon \fBcf-execd\fR is responsible for running \fBcf-agent\fR periodically.
+.br
+Documentation for CFEngine is available at http://cfengine.com/documentation/.
+.SH PROMISE THEORY
+CFEngine is built on principles from promise theory, proposed by Mark Burgess in 2004. Promise theory is a model of voluntary cooperation between individual, autonomous actors or agents who publish their intentions to one another in the form of promises. A promise is a declaration of intent whose purpose is to increase the recipient's certainty about a claim of past, present or future behaviour. For a promise to increase certainty, the recipient needs to trust the promiser, but trust can also be built on the verification that previous promises have been kept, thus trust plays a symbiotic relationship with promises.
+.br
+For an introduction to promise theory, please see http://arxiv.org/abs/0810.3294/
+.SH AVAILABILITY
+cf-support is part of CFEngine.
+.br
+Binary packages may be downloaded from http://cfengine.com/downloads/.
+.br
+The source code is available at http://github.com/cfengine/
+.SH BUGS
+Please see the public bug-tracker at http://bug.cfengine.com/.
+.br
+GitHub pull-requests may be submitted to http://github.com/cfengine/core/.
+.SH "SEE ALSO"
+.BR cf-promises (8),
+.BR cf-agent (8),
+.BR cf-serverd (8),
+.BR cf-execd (8),
+.BR cf-monitord (8),
+.BR cf-runagent (8),
+.BR cf-key (8)
+.SH AUTHOR
+Northern.tech AS
+EOF
+exit 0
+fi
+
+[ "$1" = "--yes" ] || [ "$1" = "-y" ] && non_interactive=1
+# TODO: remove set -x here
+set -e
+if [ "$EUID" -ne 0 ]; then
+  echo "$0 must be run as root"
+  exit 1
+fi
+
+
+if [ $non_interactive -eq 0 ]; then
+  read -p "If you have it, please enter your support case number: " case_number
+fi
+case_number=${case_number:-NA}
+timestamp=$(date +%F-%H%M)
+collection="cfengine_support_case_$case_number-$timestamp"
+
+export tmpdir="$(mktemp -d)/$collection"
+mkdir -p "$tmpdir"
+
+export info_file="$tmpdir/system-info.txt"
+
+function file_add
+{
+  local filename=$(basename $1)
+  if [ -f "$1" ]; then
+    cp "$1" "$tmpdir"/"$filename"
```

Comment:
maybe it's better to add `echo "adding file $1" here to be more transparent and in line with `log_cmd` which shows what commands we execute?

Reply:
mmm, an echo to show progress would be fine :+1:

Reply:
Plus we should tell use what's being grabbed from their system.